pygmt/helpers/utils.py

```diff
@@ -172,8 +172,16 @@ def build_arg_string(kwargs):
     ...     )
     ... )
     -BWSen -Bxaf -Byaf -I1/1p,blue -I2/0.25p,blue -JX4i -R1/2/3/4
+    >>> print(build_arg_string(dict(R="1/2/3/4", J="X4i", watre=True)))
+    Traceback (most recent call last):
+      ...
+    pygmt.exceptions.GMTInvalidInput: Unrecognized parameter 'watre'.
     """
     gmt_args = []
+    # raise an exception for unrecognized options
+    for key in kwargs:
+        if len(key) > 2:
+            raise GMTInvalidInput(f"Unrecognized parameter '{key}'.")
```

Comment:
Why not put this in the `for key in filtered_kwargs` loop below?

Reply:
It's OK to put it in the `for key in filtered_kwargs` loop, but it won't catch typos like `watre=False`.